File: lc_soln/0011_container_with_most_water/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int result = 0;
        while (l < r) {
            result = max(result, min(height[l], height[r]) * (r - l));
            if (height[l] < height[r]) {
                ++l;
            } else {
                --r;
            }
        }
        return result;
    }
};

int main() {
    Solution solution;
    vector<int> height= {1, 8, 6, 2, 5, 4, 8, 3, 7};
    cout << solution.maxArea(height) << "\n";
    return 0;
}

File: lc_soln/0011_container_with_most_water/solution.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int result = -1;
        int l = 0;
        int r = heights.Length - 1;
        while (l < r) {
            int h = int.Min(heights[l], heights[r]);
            int curr = (r - l) * h;
            result = int.Max(result, curr);
            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return result;
    }
}

